tutorials/get-started/index.md

```diff
@@ -0,0 +1,989 @@
+
+  [[highlight]]final _saved = new Set<WordPair>();[[/highlight]]
+
+  final TextStyle _biggerFont = new TextStyle(fontSize: 18.0);
+  ...
+}
+{% endprettify %}
+</li>
+
+<li markdown="1">
+In the `_buildRow` function, add `alreadySaved`, a check to ensure that
+a word pairing hasn't already been added to favorites.
+
+<!-- skip -->
+{% prettify dart %}
+  Widget _buildRow(WordPair pair) {
+    [[highlight]]final alreadySaved = _saved.contains(pair);[[/highlight]]
+    ...
+  }
+{% endprettify %}
+</li>
+
+<li markdown="1">
+Also in `_buildRow`, add heart-shaped icons to the ListTiles to enable
+favoriting. Display favorite entries with a solid red heart,
+and other entries with a heart-shaped outline.
+
+Make the highlighted changes below:
+
+<!-- skip -->
+{% prettify dart %}
+  Widget _buildRow(WordPair pair) {
+    final alreadySaved = _saved.contains(pair);
+    return new ListTile(
+        title: new Text(
+          pair.asPascalCase,
+          style: _biggerFont,
+        [[highlight]]),[[/highlight]]  // Replace the "));" with this.
+        [[highlight]]trailing: new Icon([[/highlight]] // Add this line, plus the next 3 lines.
+          [[highlight]]alreadySaved ? Icons.favorite : Icons.favorite_border,[[/highlight]]
+          [[highlight]]color: alreadySaved ? Colors.red : null,[[/highlight]]
+        [[highlight]]));[[/highlight]]
+  }
+}
+{% endprettify %}
+</li>
+
+<li markdown="1">
+Run the app. You should now see open hearts on each row, but they aren't yet
+interactive.
+</li>
+
+<li markdown="1"> Make the hearts tappable in the `_buildRow` function.
+If a word entry has already been added to favorites, tapping it again
+removes it from favorites. When the heart has been tapped, the function
+calls `setState()`. Calling this method signals the build system that
+state has changed and that the state needs updating.
+
+Make the highlighted changes:
+
+<!-- skip -->
+{% prettify dart %}
+  Widget _buildRow(WordPair pair) {
+    final alreadySaved = _saved.contains(pair);
+    return new ListTile(
+        title: new Text(
+          pair.asPascalCase,
+          style: _biggerFont,
+        ),
+        trailing: new Icon(
+          alreadySaved ? Icons.favorite : Icons.favorite_border,
+          color: alreadySaved ? Colors.red : null,
+        [[highlight]]),[[/highlight]] // Replace the "))," with "),"
+        [[highlight]]onTap: () {[[/highlight]] //Add this and the remaining highlighted lines.
+          [[highlight]]setState(() {[[/highlight]]
+            [[highlight]]if (alreadySaved) {[[/highlight]]
+              [[highlight]]_saved.remove(pair);[[/highlight]]
+            [[highlight]]} else {[[/highlight]]
+              [[highlight]]_saved.add(pair);[[/highlight]]
+            [[highlight]]}[[/highlight]]
+          [[highlight]]});[[/highlight]]
+        [[highlight]]});[[/highlight]]
+  }
+{% endprettify %}
+</li>
+</ol>
+
+Run the app. You should be able to tap the hearts to favorite, or unfavorite,
```

Comment:
Hot reload will work fine for the above changes. We should encourage users to hot reload whenever it's applicable and explain why they need to restart their app after making certain types of changes. Cc: @mit-mit

Reply:
Makes sense. My assumption was that "Run" was clear enough. It doesn't say "Restart". But I changed all instructions to "Hot reload" and I need to re-test to make sure what needs run. That will take me awhile.  :)